src/main/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerPanel.java

```diff
@@ -629,11 +649,11 @@ private void initializeHiddenEdges() {
 
 		// only if we don't allow sources and targets internal to paths
 		if (!_allowSourcesTargetsInPaths) {
-			// hides all incoming edges to source nodes
+			// hides all incoming directed edges to source nodes
```

Comment:
Yes that is how the code functions. What you're seeing here is the option to remove incoming edges to sources and outgoing edges from targets. I didn't remove (hide) undirected edges connected to sources/targets because they could be either incoming or outgoing